src/index.md

```diff
@@ -1445,6 +1445,48 @@ Disadvantages of this model are that the list of backports in JBS will still lis
 [To the top](#){.boxheader}
 :::
 
+# Working With Pull Requests
+
+::: {.box}
+[Quick Links]{.boxheader}
+
+* [Skara documentation on PR commands](https://wiki.openjdk.java.net/display/SKARA/Pull+Request+Commands)
+:::
+
+Once you have made a change that you want to integrate into an OpenJDK code base you need to create a _Pull Request_ (PR) on GitHub. This guide assumes that you have previous experience from using git and GitHub and won't go into details of how things work. Still, the aim is of course to provide a useful guide, so speak up if more details are needed.
+
+## Rebase before creating the PR
+
+It's likely that other people have pushed changes to the code base since you created your branch. Make sure to pull the latest changes and rebase your fix on top of that before creating your PR. This is a courtesy issue. Your reviewers shouldn't have to read your patch on top of old code that has since changed. This is hopefully obvious in cases where the upstream code has gone through cleanups or refactorings, and your patch may need similar cleanups in order to even compile. But even in cases where only smaller changes has been done, the reviewers shouldn't have to react to issues like "that line of code was moved last week, why is it back there?".
+
+~~~
+git rebase master
+~~~
+
+After the PR has been published, rebasing, force-pushing, and similar actions is strongly discouraged. Such actions will disrupt the workflow for reviewers who fetch the PR branch. Pushing new changes is fine (and even merging if necessary) for a PR under review. Incremental diffs and other tools will help your reviewers see what you have changed. In the end, all commits will be squashed into a single commit automatically, so there're actually no drawbacks what so ever to making several commits to a PR branch during review.
+
+## Life of a PR
+
+#. **Write a useful description**
+
+   The description of the PR should state what problem is being solved and shortly describe how it's solved. Reviewers and other interested readers are referred to the text in the JBS issue for details, but the description should be enough to give an overview.
+
```

Comment:
This means the JBS issue should have an evaluation so you may want to explicitly say so.

Reply:
Mostly the above test covers it. Remembering that the description is included in many emails.

Two things bug me about some PR descriptions; those that are too long and too short.
The too long ones I've seen can Include extensive performance before/after information.
They should be added in a separate comment after the description.
The too short ones are the lazy ones that just refer to the bugid.

I'm not sure where to suggest notes to this effect are inserted.

I'd also prefer to see PRs that are complete when they are created.
Code changes, test additions, copyrights updated, all done.
It is annoying to see a dribble of minor updates and accompanying emails
while the developer works through their own thought processes.
And even updating minor changes due to comments should be batched.
But this may be a developer style issue, to each their own, but why should I witness every change.

Reply:
Maybe suggest that the PR is created in `Draft` if it isn't yet complete? It won't be `rfr` and won't be sent to any manling list until it is taken out of Draft.

Reply:
I've added a sub-section to talk about things to think about before creating the PR. Feel free to add more items to that list.